colossus/src/main/scala/colossus/service/ServiceServer.scala

```diff
@@ -82,12 +84,48 @@ object ServiceConfig {
     val logErrors      = config.getBoolean("log-errors")
     val requestMetrics = config.getBoolean("request-metrics")
     val maxRequestSize = DataSize(config.getString("max-request-size"))
-    ServiceConfig(timeout, bufferSize, logErrors, requestMetrics, maxRequestSize)
+
+    val errorConf      = config.getConfig("errors")
+    val errorsDoNotLog = errorConf.getStringList("do-not-log").asScala.toSet
+    val errorsLogName  = errorConf.getStringList("log-only-name").asScala.toSet
+    ServiceConfig(timeout, bufferSize, logErrors, requestMetrics, maxRequestSize, ErrorConfig(errorsDoNotLog, errorsLogName))
   }
 }
 
+case class ErrorConfig(doNotLog: Set[String], logOnlyName: Set[String])
+
+object RequestFormatter {
+  case class LogMessage(message: String, includeStackTrace: Boolean)
+}
+
 trait RequestFormatter[I] {
-  def format(request: I): String
+  import RequestFormatter._
+  // None means do not log. Some(false) means log only the name. Some(true) means log with stack trace (default).
```

Comment:
Let's format this comment with proper scaladoc formatting so it shows up in the docs.

Personally I'd be more in favor of using a custom 3-state ADT (eg `case object DoNotLog`) which would be more descriptive than a `Option[Boolean]`, but it's not a big deal since it's only for this one method.